### DISCUSSION TITLE
solving requirement dynamism problem

### DISCUSSION THREAD

**Discussion by techoneel** [wrappid/.github#28] [Ideas]
# Solving Requirement Dynamism Problem

## Introduction

**Requirement Dynamism** is the challenge of building software in a rapidly changing business environment.
Please go through the google slide available here - https://docs.google.com/presentation/d/1dVzPCOzH11F5PT7PimzwVxsPA9Jsm5eh3qvBOsyKvPQ/edit?usp=sharing

## How to Solve

Saving meta data of the application's technical entities and business entities.

> technical entities i.e. Theme, Model, Business Entity, Route, Page, Form etc.

> business entities i.e. Organization, Appointment, Task etc.

The meta data can be stored in two particular way:

1. Technical entity meta data can be stored as JSON schema in a table
2. Business entity meta data can be stored as main table and related meta table.

## Planning

> Now to achieve this solution to be realistic the planning is important.
```
So we need to achieve it by resolving the following issues:
1. core components with it's valid props with a registry categorized
2. builder tools UI for easier creation of the technical entities and interconnectivity
3. model(table) and model meta(meta table) creation through builder UI
```

**Comment by samhere17** [comment]
@gautamdey Need your guidance here. Could you please help?
